src/router-configuration.ts

```diff
@@ -193,8 +198,9 @@ export class RouterConfiguration {
 
       let pipelineProvider = router.pipelineProvider;
       for (let i = 0, ii = pipelineSteps.length; i < ii; ++i) {
-        let {name, step} = pipelineSteps[i];
-        pipelineProvider.addStep(name, step);
+        let { name, step } = pipelineSteps[i];
+        // Potential error, addStep only accepts PipelineStep, but step here may be also a function
```

Comment:
should never be a function, as far as I know.

Reply:
It's true that it cant be a function here, the existing typing is inaccurate.